templates/federated_reporting/mrproper.py

```diff
@@ -219,26 +294,94 @@ def main():
-        feederHostname = hub["ui_name"]
-        feederApi = NovaApi(
-            api_user="mrproper",
-            api_password=mrproper_password,
-            cert_path=CERT_PATH,
-            hostname=feederHostname,
+        logger.debug(
+            "Hosts to delete on {} are {}".format(hub["ui_name"], response["rows"])
         )
+        hosts_to_delete = response["rows"]
+        if len(hosts_to_delete) != 0:
+            logger.debug(
+                "{} host(s) to delete on feeder {}".format(
+                    len(hosts_to_delete), hub["ui_name"]
+                )
+            )
+
+        # build up a post-loop SQL statement to delete hosts locally from feeder schemas
+        # change to feeder schema to make deletions easier/more direct without having to
+        # specify hub_id in queries
+        post_sql = "set schema 'hub_{}';\n".format(feeder_hubid)
+        post_sql += "\\set ON_ERROR STOP on\n"
+        delete_sql = ""
+        post_hostkeys = []
         for row in hosts_to_delete:
+            # The query API returns rows which are lists of column values.
+            # We only selected hostkey so will take the first value.
             host_to_delete = row[0]
+
+            post_hostkeys.append(host_to_delete)
+
             response = feederApi.delete("host", host_to_delete)
-            print(
-                "Delete {} on {} got response {}".format(host_to_delete, feederHostname, response)
+            if response["status"] != 202:
+                logger.warning(
+                    "Delete {} on feeder {} got {} status code".format(
+                        host_to_delete, feeder_hostname, response["status"]
+                    )
+                )
+                continue
+
+        # during this loop, also create a SQL file to execute which will
+        # simulate the host api delete process by setting current_timestamp in deleted column
+        # and delete from all federated tables similar to the clear_hosts_references() pgplsql function.
+        post_sql += "INSERT INTO __hosts (hostkey,deleted) VALUES"
+        for hostkey in post_hostkeys:
+            delete_sql += "('{}', CURRENT_TIMESTAMP) ".format(hostkey)
+        delete_sql += "ON CONFLICT (hostkey,hub_id) DO UPDATE SET deleted = excluded.deleted;\n"
+        clear_sql = "set schema 'public';\n"
+        tables = [table.strip() for table in CFE_FR_TABLES.split()]
+        for table in tables:
+            # special case of partitioning, operating on parent table will work
+            if "__promiselog_*" in table:
+                table = "__promiselog"
+            clear_sql += (
+                "DELETE FROM {} WHERE hub_id = {} AND hostkey IN ({});\n".format(
+                    table,
+                    feeder_hubid,
+                    ",".join(["'{}'".format(hk) for hk in post_hostkeys]),
+                )
             )
+        post_sql += delete_sql + clear_sql
 
+        if len(post_hostkeys) != 0:
```

Comment:
`if len(post_hostkeys) == 0` should be checked above and this whole SQL construction should be skipped in such case.

Reply:
working on refactoring this.